src/qubic.cpp

```diff
@@ -1775,6 +1776,53 @@ static void processTick(unsigned long long processorNumber)
     if (nextTickData.epoch == system.epoch)
     {
         bs->SetMem(entityPendingTransactionIndices, sizeof(entityPendingTransactionIndices), 0);
+        // reset solution task queue
+        score->resetTaskQueue();
+        // pre-scan any solution tx and add them to solution task queue
+        for (unsigned int transactionIndex = 0; transactionIndex < NUMBER_OF_TRANSACTIONS_PER_TICK; transactionIndex++)
+        {
+            if (!isZero(nextTickData.transactionDigests[transactionIndex]))
+            {
+                if (tickTransactionOffsets[system.tick - system.initialTick][transactionIndex])
+                {
+                    Transaction* transaction = (Transaction*)&tickTransactions[tickTransactionOffsets[system.tick - system.initialTick][transactionIndex]];
+                    const int spectrumIndex = ::spectrumIndex(transaction->sourcePublicKey);
+                    if (spectrumIndex >= 0
+                        && !entityPendingTransactionIndices[spectrumIndex])
+                    {
+                        if (transaction->destinationPublicKey == arbitratorPublicKey)
+                        {
+                            if (!transaction->amount
+                                && transaction->inputSize == 32
+                                && !transaction->inputType)
+                            {
+                                const m256i& solution_nonce = *(m256i*)((unsigned char*)transaction + sizeof(Transaction));
+                                m256i data[2] = { transaction->sourcePublicKey, solution_nonce };
+                                static_assert(sizeof(data) == 2 * 32, "Unexpected array size");
+                                unsigned int flagIndex;
+                                KangarooTwelve(data, sizeof(data), &flagIndex, sizeof(flagIndex));
+                                if (!(minerSolutionFlags[flagIndex >> 6] & (1ULL << (flagIndex & 63))))
+                                {
+                                    score->addTask(transaction->sourcePublicKey, solution_nonce);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        {
+            // start processing solutions in this tick

```

Comment:
I would add a bit more info here. Like "tryProcessSolution() is called by requestProcessors in parallel to speed up solution processing"